treeTable.js

```diff
@@ -59,39 +59,26 @@
         if (key === 0) {
             return 1
         }
-        const parentKey = self.data.filter((d) => d.tt_key === key)[0].tt_parent;
-        return 1 + level(self, parentKey);
-    }
-
-    function hasChild (self, key) {
```

Comment:
This was being called twice, now creating a dictionary of keys to data on construction with children stored as part of the data objects.